src/PAModel/Serializers/MsAppSerializer.cs

```diff
@@ -537,14 +537,22 @@ private static IEnumerable<FileEntry> GetMsAppFiles(this CanvasDocument app, Err
                 yield return ToFile(FileKind.PublishInfo, publishInfo);
 
             // "DataComponent" data sources are not part of DataSource.json, and instead in their own file
+            // In case the the entropy is missing then make sure that tableDefinitions are being added towards the end in DataSources.json file
+            // This is to respect the serialization order of the server.
             var dataSources = new DataSourcesJson
             {
                 DataSources = app.GetDataSources()
                     .SelectMany(x => x.Value)
-                    .Where(x => !x.IsDataComponent)
+                    .Where(x => !x.IsDataComponent && x.TableDefinition == null)
```

Comment:
Instead of doing this in two passes, you could also put Int.MaxValue for the ones with table definitions in the OrderBy statement

Reply:
test...

Reply:
@MikeStall this change was reverted and a test case was added for this, do you want to add another test for something?